docker/zap_common.py

```diff
@@ -382,23 +387,32 @@ def zap_spider(zap, target):
 
 
 @hook(wrap=True)
-def zap_ajax_spider(zap, target, max_time):
+def zap_ajax_spider(zap, target, max_time, context=None):
     logging.debug('AjaxSpider ' + target)
     if max_time:
         zap.ajaxSpider.set_option_max_duration(str(max_time))
-    zap.ajaxSpider.scan(target)
+    if context is not None:
+        spider_scan_id = zap.spider.scan(target, contextname=context)
+    else:
+        spider_scan_id = zap.spider.scan(target)
     time.sleep(5)
 
     while (zap.ajaxSpider.status == 'running'):
         logging.debug('Ajax Spider running, found urls: ' + zap.ajaxSpider.number_of_results)
         time.sleep(5)
     logging.debug('Ajax Spider complete')
 
-
+'''
+In zap_active_scan, we use the global context_id, which is used in zap.ascan.scan to indicate that there is
+a context
+'''
 @hook(wrap=True)
 def zap_active_scan(zap, target, policy):
     logging.debug('Active Scan ' + target + ' with policy ' + policy)
-    ascan_scan_id = zap.ascan.scan(target, recurse=True, scanpolicyname=policy)
+    if context_id is None:
+        ascan_scan_id = zap.ascan.scan(target, recurse=True, scanpolicyname=policy)#, contextid=None)
```

Comment:
This looks a bit weird. Cant we just use contextid=context_id and drop the test?

Reply:
I think we could use **kwargs since the argument "contextid" can appear or not. This way the test could be dropped..

Reply:
We could use **kwargs in the zap_active_scan function?

Reply:
the context_id variable is initialized in the zap_import_context function.. this function is only called if a context_file is given. So context_id is not always available..

Reply:
It's fine if it's `None`, the API client will just ignore that parameter.